From 4210ac27b86ea0aee325c9be1603db5254354b1a Mon Sep 17 00:00:00 2001
From: Pavithra R <pavir@codeaurora.org>
Date: Thu, 17 Dec 2020 10:15:26 +0530
Subject: [PATCH] testmgr: Disable zero input data length for HASH

This patch is a manual propagation of commit
996453649de0bac629c6dd9f530d6bd7b07c1558.
Zero input data length for HASH computation is not supported in the
current IP release. Hence disabling the use-case only for
ipq50xx.

Change-Id: I4a4a42db7e10b25d47325530dfca1fd3af292618
Signed-off-by: Pavithra R <pavir@codeaurora.org>
---
 crypto/Kconfig   |  7 +++++++
 crypto/testmgr.h | 16 ++++++++++++++--
 2 files changed, 21 insertions(+), 2 deletions(-)

--- a/crypto/Kconfig
+++ b/crypto/Kconfig
@@ -256,6 +256,13 @@ config CRYPTO_DISABLE_AES192_TEST
 	help
 	  Disable testcases with 192 key length
 
+config CRYPTO_NO_ZERO_LEN_HASH
+	bool "Disable support for zero data length cases"
+	default y
+	depends on CRYPTO_TEST
+	help
+	  Disable support for zero data length cases
+
 config CRYPTO_SIMD
 	tristate
 	select CRYPTO_CRYPTD
--- a/crypto/testmgr.h
+++ b/crypto/testmgr.h
@@ -4270,12 +4270,18 @@ static const struct hash_testvec hmac_sm
  * Long vector from CAVS 5.0
  */
 static const struct hash_testvec sha1_tv_template[] = {
+#ifndef CONFIG_CRYPTO_NO_ZERO_LEN_HASH
+	/*
+	 * zero-length input data length is not supported in ipq50xx
+	 */
 	{
 		.plaintext = "",
 		.psize	= 0,
 		.digest	= "\xda\x39\xa3\xee\x5e\x6b\x4b\x0d\x32\x55"
 			  "\xbf\xef\x95\x60\x18\x90\xaf\xd8\x07\x09",
-	}, {
+	},
+#endif
+	{
 		.plaintext = "abc",
 		.psize	= 3,
 		.digest	= "\xa9\x99\x3e\x36\x47\x06\x81\x6a\xba\x3e"
@@ -4626,6 +4632,10 @@ static const struct hash_testvec sha224_
  * SHA256 test vectors from NIST
  */
 static const struct hash_testvec sha256_tv_template[] = {
+#ifndef CONFIG_CRYPTO_NO_ZERO_LEN_HASH
+	/*
+	 * zero-length input data length is not supported in ipq50xx
+	 */
 	{
 		.plaintext = "",
 		.psize	= 0,
@@ -4633,7 +4643,9 @@ static const struct hash_testvec sha256_
 			  "\x9a\xfb\xf4\xc8\x99\x6f\xb9\x24"
 			  "\x27\xae\x41\xe4\x64\x9b\x93\x4c"
 			  "\xa4\x95\x99\x1b\x78\x52\xb8\x55",
-	}, {
+	},
+#endif
+	{
 		.plaintext = "abc",
 		.psize	= 3,
 		.digest	= "\xba\x78\x16\xbf\x8f\x01\xcf\xea"
